packages/esm-appointments-app/src/change-appointment-status/change-appointment-status.component.tsx

```diff
@@ -0,0 +1,103 @@
+  closeModal: () => void;
+}
+
+const ChangeAppointmentStatusModal: React.FC<ChangeAppointmentStatusModalProps> = ({ appointments, closeModal }) => {
+  const { t } = useTranslation();
+  const { mutate } = useSWRConfig();
+  const { name, id, dateTime, serviceType, status } = appointments;
+  const [selectedStatus, setSelectedStatus] = useState(status);
+  const appointmentStatus = [
+    { display: t('checkedIn', 'CheckedIn'), value: 'CheckedIn' },
+    { display: t('missed', 'Missed'), value: 'Missed' },
+    { display: t('completed', 'Completed'), value: 'Completed' },
+  ];
+
+  const handleChangeAppointmentStatus = async () => {
+    const ac = new AbortController();
+    const { status, statusText } = await changeAppointmentStatus(selectedStatus, id, ac);
+    status === 200
+      ? showToast({
+          critical: true,
+          kind: 'success',
+          description: t(
+            'appointmentStatusChange',
+            `Appointment status has been successfuly changed to ${selectedStatus}`,
+          ),
+          title: t('appointmentStatusTitleMessage', 'Appointment status'),
+        })
+      : showNotification({
+          title: t('appointmentFormError', 'Error updating appointment status'),
+          kind: 'error',
+          critical: true,
+          description: statusText,
+        });
+    selectedStatus === 'CheckedIn' && navigate({ to: `\${openmrsSpaBase}/outpatient` });
+    mutate(`/ws/rest/v1/appointment/appointmentStatus?forDate=${startDate}&status=Scheduled`);
+    closeModal();
+  };
+
+  return (
+    <div>
+      <ModalHeader closeModal={closeModal} title={t('changeAppointmentStatus', 'Change appointment status')} />
+      <ModalBody>
+        <div className={styles.modalBody}>
+          <h4>
+            {t('patientName', 'Patient name')}&nbsp;
+            {name}
+          </h4>
+          <h5>
+            {t('serviceType', 'Service type')}&nbsp;
+            {serviceType}
+          </h5>
+          <h6>
+            {t('startTime', 'Start time')}&nbsp;
+            {dateTime}
+          </h6>
+        </div>
+        <Form onSubmit={() => {}}>
+          <FormGroup legendText="">
+            <RadioButtonGroup
+              className={styles.radioButtonGroup}
+              valueSelected={selectedStatus}
+              orientation="vertical"
+              onChange={(value) => setSelectedStatus(value)}
+              name="radio-button-group">
+              {appointmentStatus.map(({ value, display }) => (
+                <RadioButton key={value} className={styles.radioButton} id={value} labelText={display} value={value} />
+              ))}
+            </RadioButtonGroup>
+          </FormGroup>
+        </Form>
+      </ModalBody>
+      <ModalFooter>
+        <Button kind="secondary" onClick={closeModal}>
+          {t('cancel', 'Cancel')}
+        </Button>
+        <Button onClick={handleChangeAppointmentStatus}>{t('exitAndChangeStatus', 'Exit and change status')}</Button>
```

Comment:
When i first saw `Exit` i quickly looked for another button for saving but instead saw `Cancel`. Then i came back and noticed that it is `Exit` and then `change status`. I guess you now understand what am saying. 😊

Reply:
And on another note, i have seen this pattern all over 03. In the old days, such a window would simply have `Change status` or `Save status` without `Exit`. And it was the trend that the popup window would automatically close after clicking any of the Save or Cancel buttons. Exit was mostly reserved to exiting the entire application. `Close` would be for such small popups, if we had to. Could this be a sign of my old age and out of touch with the new modernity? 😊